modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/model.scala

```diff
@@ -289,10 +304,28 @@ object Step {
   * the `Queue` for proper insertion into the completed `Queue` when all the
   * `Execution`s are done.
   */
-case class Current(actions: Vector[Action \/ Result],
+case class Current(ars: Vector[Action \/ Result],
                    // TODO: The following tuples should be replaced by either
                    // Sequence or Step parameters
-                   ctxt: Option[(String, Int) \/ Int])
+                   ctxt: Option[(String, Int) \/ Int]) {
+
+  def isEmpty: Boolean = this.ars.empty
+
+  def actions: List[Action] = {
+
+    // not available in scalaz?
+    def lefts[L, R](xs: List[L \/ R]): List[L] = for { -\/(r) <- xs } yield r
```

Comment:
I was thinking about the [Haskell `lefts`](https://hackage.haskell.org/package/base-4.9.0.0/docs/Data-Either.html#v:lefts). `leftMap` maps over every `left` but I just want to *collect* every left in a list of `\/`. I'm pretty sure there'll be something in scalaz to do that but can't find how it's named.